src/crimson/os/seastore/cached_extent.h

```diff
@@ -591,18 +638,33 @@ class CachedExtent : public boost::intrusive_ref_counter<
 protected:
   CachedExtent(CachedExtent &&other) = delete;
   CachedExtent(ceph::bufferptr &&ptr) : ptr(std::move(ptr)) {}
-  CachedExtent(const CachedExtent &other)
-    : state(other.state),
-      dirty_from_or_retired_at(other.dirty_from_or_retired_at),
-      ptr(other.ptr.c_str(), other.ptr.length()),
-      version(other.version),
-      poffset(other.poffset) {}
+  CachedExtent(extent_len_t length) : ptr(ceph::bufferptr(
+    buffer::create_page_aligned(length))) {}
```

Comment:
Is this new Constructor necessary? It seems a little bit similar to `CachedExtent::(ceph::bufferptr &&)`, maybe only one of them needs to be preserved?

Reply:
Who is the user of this new constructor?

Reply:
It is prepared for all types of CachedExtent other than ObjectBlock. Since I modify the constructor in get_extent, I add the new constructor in case. The old one is still used in some places. If we ensure CachedExtent(extent_len_t length) is used only for ObjectBlock or modify the CachedExtent(ceph::bufferptr &&ptr) in other places, only one of them is needed. (I haven't checked them)

Reply:
Just realized where `CachedExtent(extent_len_t)` is used, refer to the comment https://github.com/ceph/ceph/pull/50421#discussion_r1145712882